src/utils.rs

```diff
@@ -0,0 +1,8 @@
+use crate::{exceptions, PyErr};
```

Comment:
I'm personally not a fan of "utils" files if we can have code in more precisely named files.

In this case, I think once the `buffer.rs` changes are reverted then this function is only used inside `conversions/array.rs`, so can be moved there.